src/index/TextIndexBuilder.cpp

```diff
@@ -135,47 +136,69 @@ void TextIndexBuilder::processWordsForInvertedLists(
 }
 
 // _____________________________________________________________________________
-cppcoro::generator<WordsFileLine> TextIndexBuilder::wordsInTextRecords(
-    std::string contextFile, bool addWordsFromLiterals) const {
-  auto localeManager = textVocab_.getLocaleManager();
+ad_utility::InputRangeTypeErased<WordsFileLine>
+TextIndexBuilder::wordsInTextRecords(std::string contextFile,
+                                     bool addWordsFromLiterals) const {
   // ROUND 1: If context file aka wordsfile is not empty, read words from there.
   // Remember the last context id for the (optional) second round.
-  TextRecordIndex contextId = TextRecordIndex::make(0);
-  if (!contextFile.empty()) {
-    WordsFileParser p(contextFile, localeManager);
-    ad_utility::HashSet<std::string> items;
-    for (auto& line : p) {
-      contextId = line.contextId_;
-      co_yield line;
-    }
-    if (contextId > TextRecordIndex::make(0)) {
-      contextId = contextId.incremented();
+  std::shared_ptr<LocaleManager> localeManager =
+      std::make_shared<LocaleManager>(textVocab_.getLocaleManager());
+  std::shared_ptr<TextRecordIndex> nextContextId =
+      std::make_shared<TextRecordIndex>(TextRecordIndex::make(0));
```

Comment:
Use `auto [varName] = std::make_shared<....>(...)`
(less redundant + hopefully fits in one line.